apps/meteor/client/views/admin/apps/components/ScreenshotCarousel.tsx

```diff
@@ -0,0 +1,117 @@
+import { Box, Button, Icon } from '@rocket.chat/fuselage';
+import colors from '@rocket.chat/fuselage-tokens/colors';
+import React, { ReactElement } from 'react';
+
+import { Screenshot } from '../definitions/AppInfo';
+
+type ScreenshotCarouselProps = {
+	AppScreenshots: Array<Screenshot>;
+	setViewCarousel: React.Dispatch<React.SetStateAction<boolean>>;
+	handleNextSlide: () => void;
+	handlePrevSlide: () => void;
+	isFirstSlide: boolean;
+	isLastSlide: boolean;
+	currentSlideIndex: number;
+};
+
+const ScreenshotCarousel = ({
+	AppScreenshots,
+	setViewCarousel,
+	handleNextSlide,
+	handlePrevSlide,
+	isFirstSlide,
+	isLastSlide,
+	currentSlideIndex,
+}: ScreenshotCarouselProps): ReactElement => (
+	<>
+		<Box position='fixed' w='100%' h='100vh' bg={colors.n800} opacity='0.7' zIndex='2' marginBlock='-0.75px' />
+
+		{!isFirstSlide && (
+			<Button
+				onClick={handlePrevSlide}
+				display='flex'
+				alignItems='center'
+				justifyContent='center'
+				style={{ top: '50%', left: '10px', cursor: 'pointer', transform: 'translateY(-50%)' }}
+				position='absolute'
+				zIndex={3}
+				borderRadius='x2'
+				w='x28'
+				h='x28'
+				margin='0'
+				bg={colors.n600}
+				borderColor={colors.n600}
+			>
+				<Icon name='chevron-right' size='x24' color='alternative' />
+			</Button>
+		)}
+
+		{!isLastSlide && (
+			<Button
+				onClick={handleNextSlide}
+				display='flex'
+				alignItems='center'
+				justifyContent='center'
+				style={{ top: '50%', right: '10px', cursor: 'pointer', transform: 'translateY(-50%)' }}
+				position='absolute'
+				zIndex={3}
+				borderRadius='x2'
+				w='x28'
+				h='x28'
+				margin='0'
+				bg={colors.n600}
+				borderColor={colors.n600}
+			>
+				<Icon name='chevron-left' size='x24' color='alternative' />
+			</Button>
+		)}
+
+		<Box
+			onClick={(): void => setViewCarousel(false)}
+			position='fixed'
+			width='100%'
+			height='100vh'
+			display='flex'
+			justifyContent='center'
+			alignItems='center'
+			zIndex='2'
+			mi='x38'
+		>
+			{AppScreenshots.map((currentScreenshot, index) => {
```

Comment:
Maybe you can refactor this area in a getContent function, or break the component for screenshots in other components and use this inside of this map.

Reply:
Solved